_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -110,6 +106,176 @@ unsigned int ConvertMfxFourccToVAFormat(mfxU32 fourcc)
     }
 }
 
+static mfxU32 FillSurfaceAttrs(VASurfaceAttrib *attrib, unsigned int *format,  mfxU32 fourcc, mfxU32 va_fourcc, mfxU32 memType)
+{
+    mfxU32 attrCnt=0;
+
+    attrib[attrCnt].type            = VASurfaceAttribPixelFormat;
```

Comment:
please don't do it this way. You are owning some pointer coming from outside and iterate through it assuming that there is enough memory. It could lead to heap corruption. I still insist to use vector